### PR TITLE
Feature/adminitrative dashboard

### PR DESCRIPTION
**Pull request by xkevin190**
a hidden administration panel was added, with additional functionalities.

### CODE REVIEW

File: android/app/src/main/java/io/locha/p2p/runtime/ChatServiceModule.java [hunk 1]
```diff
@@ -150,11 +154,14 @@ public void getPeerId(Promise promise) {
      *
      * @see <a href="https://multiformats.io/multiaddr/">Multiaddr</a>
      */
-    @ReactMethod public void dial(String multiaddr) {
+    @ReactMethod public void dial(String multiaddr, Promise promise) {
+        Log.i(TAG, "dial: hereeeee");
```
Comment (jeandudey) [inline android/app/src/main/java/io/locha/p2p/runtime/ChatServiceModule.java:158]:
```suggestion
        Log.i(TAG, String.format("dialing \"%s\"", multiaddr);
```

File: android/app/src/main/java/io/locha/p2p/runtime/ChatServiceModule.java [hunk 2]
```diff
@@ -150,11 +154,14 @@ public void getPeerId(Promise promise) {
...
         try {
             nativeDial(multiaddr);
+            promise.resolve(null);
         } catch (Exception e){
-            Log.e(TAG, "Couldn't dial address", e);
+            Log.e(TAG, "Couldn't dial address"+ e.toString());
```
Comment (jeandudey) [inline android/app/src/main/java/io/locha/p2p/runtime/ChatServiceModule.java:163]:
```suggestion
            Log.e(TAG, "Couldn't dial address", e);
```

This way logcat will show the stack trace too. `nativeDial` shouldn't fail, only when ChatService hasn't been started, any other error is reported though events.

File: android/app/src/main/java/io/locha/p2p/runtime/EventsReceiver.java [hunk 3]
```diff
@@ -86,7 +86,7 @@ public static EventsReceiver get() {
     }
 
     @Override public void onIncomingConnectionError(String localAddr, String sendBackAddr, String error) {
-        Log.d(TAG, String.format("incomingConnection: localAddr=%s, sendBackAddr=%s, error=%s",
+        Log.d(TAG, String.format("incomingConnection Error: localAddr=%s, sendBackAddr=%s, error=%s",
```
Comment (jeandudey) [inline android/app/src/main/java/io/locha/p2p/runtime/EventsReceiver.java:89]:
```suggestion
        Log.d(TAG, String.format("incomingConnectionError: localAddr=%s, sendBackAddr=%s, error=%s",
```

File: src/store/chats/chatAction.js [hunk 4]
```diff
@@ -354,3 +354,76 @@ export const sendAgain = (message) => (dispatch) => {
...
+export const startManualService = (callback) => async (dispatch) => {
+  try {
+    await chatService.startService();
+    callback();
+    dispatch({
+      type: ActionTypes.CHAT_SERVICE_STATUS,
+      payload: true
+    });
+  } catch (error) {
+    console.log('error to the start chat service');
```
Comment (jeandudey) [inline src/store/chats/chatAction.js:372]:
```suggestion
    console.log('couldn't start chat service');
```

File: android/app/src/main/java/io/locha/p2p/runtime/ChatServiceModule.java (hunk 2 above)
Comment (luisan00) [reply android/app/src/main/java/io/locha/p2p/runtime/ChatServiceModule.java:163]: other string can be:
```java
 Log.e(TAG, "Couldn't dial la verga huevona", e)
```
:smile:

File: src/store/chats/chatAction.js [hunk 5]
```diff
@@ -354,3 +354,76 @@ export const sendAgain = (message) => (dispatch) => {
...
+  try {
+    await chatService.stop();
+    console.log('despues del maldito servicio de mierda');
+    callback();
+    dispatch({
+      type: ActionTypes.CHAT_SERVICE_STATUS,
+      payload: false
+    });
+  } catch (error) {
+    console.log('error to the stop chat service', error);
```
Comment (luisan00) [inline src/store/chats/chatAction.js:391]:
```suggestion
    console.log("couldn't stop chat service: ", error);
```

File: src/views/config/AdministrativeComponent.js [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+      this.props.openAdministrativePanel(() => {
+        toast('admin panel is no longer hidden');
+      });
+    }
+  }
+
+  startOrClose = async () => {
+    if (this.props.chatService) {
+      this.props.stopService(() => {
+        toast('Chat service it stoped ');
```
Comment (luisan00) [inline src/views/config/AdministrativeComponent.js:49]:
```suggestion
        toast('Chat service is stopped ');
```

File: src/views/config/AdministrativeComponent.js [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+      });
+    } else {
+      this.props.startManualService(() => {
+        toast('chat service is started successfully');
```
Comment (luisan00) [inline src/views/config/AdministrativeComponent.js:53]:
```suggestion
        toast('Chat service started successfully');
```

**Review by luisan00 — CHANGES_REQUESTED** [review]

File: src/store/chats/chatAction.js [hunk 8]
```diff
@@ -354,3 +354,76 @@ export const sendAgain = (message) => (dispatch) => {
...
+
+/**
+ * manual stop of the chat service
+ *
+ * @brief this function is executed jus only in the administrative panel
+ */
+export const stopService = (callback) => async (dispatch) => {
+  try {
+    await chatService.stop();
+    console.log('despues del maldito servicio de mierda');
```
[the remark this answers is not in the document]
Comment (xkevin190) [reply src/store/chats/chatAction.js:384]: upsy! i think i was mad when i wrote that log  xD

**Review by jeandudey — APPROVED** [review]

File: src/views/config/AdministrativeComponent.js [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+        dialAddress: false
+      });
+    } else {
+      this.setState({
+        manualBootstrap: false
+      });
+    }
+  }
+
+
```
Comment (luisan00) [inline src/views/config/AdministrativeComponent.js:70]:
```suggestion
```

**Review by luisan00 — APPROVED** [review]

**Pull request merged**